godot-core/src/obj/guards.rs

```diff
@@ -81,65 +83,105 @@ impl<T: GodotClass> Drop for GdMut<'_, T> {
 
 // ----------------------------------------------------------------------------------------------------------------------------------------------
 
-/// Shared reference guard for a [`Base`](crate::obj::Base) pointer.
-///
-/// This can be used to call methods on the base object of a rust object that take `&self` as the receiver.
-///
-/// See [`WithBaseField::base()`](super::WithBaseField::base()) for usage.
-pub struct BaseRef<'a, T: GodotClass> {
-    gd: Gd<T::Base>,
-    _instance: &'a T,
-}
+macro_rules! make_base_ref {
+    ($ident: ident, $bound: tt, $doc_type: ident, $doc_path: path, $object_name: literal) => {
```

Comment:
`$bound` could probably be `:ty` or `:ident`, no?

Also please omit spaces after colon, i.e. `$ident:ident` -- this follows [Rust guide](https://doc.rust-lang.org/book/ch19-06-macros.html).